iina/PlayerWindowController.swift

```diff
@@ -0,0 +1,170 @@
+//
+//  PlayerWindowController.swift
+//  iina
+//
+//  Created by Yuze Jiang on 2/15/20.
+//  Copyright © 2020 lhc. All rights reserved.
+//
+
+import Foundation
+
+class PlayerWindowController: NSWindowController {
+  unowned var player: PlayerCore
+  
+  var menuActionHandler: MainMenuActionHandler!
+  
+  var loaded = false
+  
+  init(playerCore: PlayerCore) {
+    self.player = playerCore
+    super.init(window: nil)
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  internal lazy var followGlobalSeekTypeWhenAdjustSlider: Bool = Preference.bool(for: .followGlobalSeekTypeWhenAdjustSlider)
+  
+  /** Observers added to `UserDefauts.standard`. */
+  internal var notificationObservers: [NotificationCenter: [NSObjectProtocol]] = [:]
+  
+  @IBOutlet weak var volumeSlider: NSSlider!
+  @IBOutlet weak var muteButton: NSButton!
+  @IBOutlet weak var playButton: NSButton!
+  @IBOutlet weak var playSlider: NSSlider!
+  @IBOutlet weak var rightLabel: DurationDisplayTextField!
+  @IBOutlet weak var leftLabel: NSTextField!
+  
+  override func windowDidLoad() {
+    super.windowDidLoad()
+    loaded = true
```

Comment:
Is this different than [`NSWindowController.isWindowLoaded`](https://developer.apple.com/documentation/appkit/nswindowcontroller/1527496-iswindowloaded)?

Reply:
This variable is added by me recently, see f9b23d4. Should be the same, but accessing `NSWindowController.isWIndowLoaded` requires main thread, but we have too many of them that we cannot ensure they are all on the main thread. Any idea?

Reply:
Hmm, I see. It's somewhat annoying to have to keep track of this ourselves, but for now I don't have any ideas.